backend/src/main/java/com/team4/backend/util/DateUtil.java

```diff
@@ -0,0 +1,26 @@
+package com.team4.backend.util;
+
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public abstract class DateUtil {
+
+    public static LocalDate retrieveDateClosestToToday(Set<LocalDate> localDateSet) {
+        LocalDate currentDate = LocalDate.now();
+        Set<LocalDate> dateComingBeforeToday = localDateSet
```

Comment:
Not need to change, just a comment.
stream().reduce() could've been used to find the date with the minimum "distance" with the current date

Reply:
yeah you might be right i'm going to refactor it

Reply:
but it is also tricky because i have to prioritize date that comes after so i think it is much simpler my but i am interessted to see your way, we can talk about it friday